backend/compact-connect/lambdas/provider-data-v1/tests/function/test_data_model/test_client.py

```diff
@@ -0,0 +1,218 @@
+        # Verify that we're getting the expected provider ID
+        self.assertEqual(expected_provider_id, resp)
+
+    def test_get_provider_id_not_found(self):
+        """
+        Provider ID not found should raise an exception
+        """
+        from data_model.client import DataClient
+        from exceptions import CCNotFoundException
+
+        client = DataClient(self.config)
+
+        # This SSN isn't in the DB, so it should raise an exception
+        with self.assertRaises(CCNotFoundException):
+            client.get_provider_id(compact='aslp', ssn='321-21-4321')  # pylint: disable=unexpected-keyword-arg
+
+    def test_get_provider(self):
+        from data_model.client import DataClient
+
+        provider_id = self._load_provider_data()
+
+        client = DataClient(self.config)
+
+        resp = client.get_provider(  # pylint: disable=missing-kwoa,unexpected-keyword-arg
+            compact='aslp',
+            provider_id=provider_id
+        )
+        self.assertEqual(3, len(resp['items']))
+        # Should be one each of provider, license, privilege
+        self.assertEqual({'provider', 'license', 'privilege'}, {record['type'] for record in resp['items']})
+
+    def test_get_provider_garbage_in_db(self):
+        """
+        Because of the risk of exposing sensitive data to the public if we manage to get corrupted
+        data into our database, we'll specifically validate data coming _out_ of the database
+        and throw an error if it doesn't look as expected.
+        """
+        from data_model.client import DataClient
+        from exceptions import CCInternalException
+
+        provider_id = self._load_provider_data()
+
+        with open('tests/resources/dynamo/license.json', 'r') as f:
+            license_record = json.load(f)
+
+        self._table.put_item(
+            Item={
+                # Oh, no! We've somehow put somebody's SSN in the wrong place!
+                'something_unexpected': '123-12-1234',
+                **license_record
+            }
+        )
+
+        client = DataClient(self.config)
+
+        # This record should not be allowed out via API
+        with self.assertRaises(CCInternalException):
+            client.get_provider(  # pylint: disable=missing-kwoa,unexpected-keyword-arg
+                compact='aslp',
+                provider_id=provider_id
+            )
+
+    def test_get_licenses_sorted_by_family_name(self):
+        from data_model.client import DataClient
+
+        self._generate_providers(home='oh', privilege='ne', start_serial=9999)
+        self._generate_providers(home='ne', privilege='oh', start_serial=9989)
+        self._generate_providers(home='ne', privilege='co', start_serial=9979)
+        client = DataClient(self.config)
+
+        # We expect to see 20 providers: 10 have privileges in oh, 10 have licenses in oh
+        resp = client.get_licenses_sorted_by_family_name(  # pylint: disable=unexpected-keyword-arg,missing-kwoa
```

Comment:
Oh, that's a good point. I didn't rename this method when I updated it to fetch 'providers' (really a collection of 'provider', 'license', and 'privilege' records), rather than just licenses.

Reply:
Got it. So I assume that means we can rename this function to `get_providers_sorted_by_family_name`?

Reply:
Yes